sklearn/pipeline.py

```diff
@@ -37,6 +39,32 @@
 __all__ = ["Pipeline", "FeatureUnion", "make_pipeline", "make_union"]
 
 
+@contextmanager
+def _handle_warnings(estimator):
+    """A context manager to make sure a NotFittedError is raised, if a subestimator
+    raises the error.
+
+    Otherwise, we raise a warning if the pipeline is not fitted, with the deprecation.
```

Comment:
This context manager raises a warning instead of a NotFittedError, which differs from what is written here.

Maybe like this:
```suggestion
    """A context manager to raise a FutureWarning during the deprecation period, 
    if the last step of a pipeline raises a NotFittedError when it is not fitted.
```

Reply:
Now I see I got this wrong.
Would still be good to explain better what is supposed to happen.

Reply:
I think for a helper method which is only here for two versions during the deprecation cycle, it doesn't really matter.